inbm/dispatcher-agent/dispatcher/source/ubuntu_source_manager.py

```diff
@@ -98,11 +102,19 @@ def __init__(self) -> None:
 
     def add(self, parameters: ApplicationAddSourceParameters) -> None:
         """Adds a source file and optional GPG key to be used during Ubuntu application updates."""
-        # Step 1: Add key (Optional)
+        # Step 1: Verify gpg key uri from trusted repo list 
+        try:
+            url = parameters.gpg_key_uri
+            #URL slicing to remove the last segment (filename) from the URL 
+            source = url.value[:-(len(url.value.split('/')[-1]) + 1)]
+            verify_source(source=source, dispatcher_broker=dispatcher_broker)
+        except DispatcherException as err:
+            raise SourceError("Source Gpg key URI verification check failed")
+        # Step 2: Add key (Optional)
```

Comment:
Update the comment to reflect the new step numbers after adding the verification step. [medium]